### PR TITLE
chore: remove args files

### PR DESCRIPTION
**Pull request by slandelle**
Motivation:

We don't need an external temporary file, we can directly ship it inside the booter jar.

Note: this is how I already implemented this for scala-maven-plugin.

### CODE REVIEW

File: src/main/java/io/gatling/plugin/util/Fork.java [hunk 1]
```diff
@@ -223,37 +218,43 @@ public void run() throws Exception {
...
-
-      manifest.getMainAttributes().putValue(Attributes.Name.MANIFEST_VERSION.toString(), "1.0");
-      manifest.getMainAttributes().putValue(Attributes.Name.CLASS_PATH.toString(), cp.toString());
-      manifest.getMainAttributes().putValue(Attributes.Name.MAIN_CLASS.toString(), startClassName);
-      manifest.getMainAttributes().putValue(GATLING_MANIFEST_VALUE, "true");
-
+      JarEntry manifestJarEntry = new JarEntry("META-INF/MANIFEST.MF");
+      jos.putNextEntry(manifestJarEntry);
       manifest.write(jos);
+      jos.closeEntry();
```
Comment (Isammoc) [inline src/main/java/io/gatling/plugin/util/Fork.java:248]: shouldn't we always close the entry with a try/finally?

File: src/main/java/io/gatling/plugin/util/Fork.java [hunk 2]
```diff
@@ -223,37 +218,43 @@ public void run() throws Exception {
...
+
+      JarEntry argsFileEntry = new JarEntry(ForkMain.ARGS_FILE);
+      jos.putNextEntry(argsFileEntry);
+      byte[] argsBytes =
+          args.stream()
+              .collect(Collectors.joining(System.lineSeparator()))
+              .getBytes(StandardCharsets.UTF_8);
+      jos.write(argsBytes);
+      jos.closeEntry();
```
Comment (Isammoc) [inline src/main/java/io/gatling/plugin/util/Fork.java:257]: same about try/finally block

File: src/main/java/io/gatling/plugin/util/Fork.java [hunk 3]
```diff
@@ -223,37 +218,43 @@ public void run() throws Exception {
...
+      JarEntry manifestJarEntry = new JarEntry("META-INF/MANIFEST.MF");
+      jos.putNextEntry(manifestJarEntry);
       manifest.write(jos);
+      jos.closeEntry();
+
+      JarEntry argsFileEntry = new JarEntry(ForkMain.ARGS_FILE);
+      jos.putNextEntry(argsFileEntry);
+      byte[] argsBytes =
+          args.stream()
+              .collect(Collectors.joining(System.lineSeparator()))
```
Comment (Isammoc) [inline src/main/java/io/gatling/plugin/util/Fork.java:254]: System dependent line separator or a blind `\n`?

File: src/main/java/io/gatling/plugin/util/ForkMain.java [hunk 4]
```diff
@@ -19,24 +19,41 @@
...
+      runMain(cl, mainClassName, argsFromFile);
     } catch (Throwable t) {
       t.printStackTrace();
       System.exit(-1);
     }
   }
 
-  private static void runMain(String mainClassName, List<String> args) throws Exception {
-    ClassLoader cl = Thread.currentThread().getContextClassLoader();
+  private static List<String> readArgFile(ClassLoader cl) throws IOException {
```
Comment (Isammoc) [inline src/main/java/io/gatling/plugin/util/ForkMain.java:42]: name is `readArgFile` but we don't have file anymore?

File: src/main/java/io/gatling/plugin/util/Fork.java [hunk 5]
```diff
@@ -223,37 +218,43 @@ public void run() throws Exception {
...
+        classPath.stream()
+            .map(element -> getURL(new File(element)).toExternalForm())
+            .collect(Collectors.joining(" "));
+    Manifest manifest = new Manifest();
+    manifest.getMainAttributes().putValue(Attributes.Name.MANIFEST_VERSION.toString(), "1.0");
+    manifest.getMainAttributes().putValue(Attributes.Name.CLASS_PATH.toString(), cp);
+    manifest
+        .getMainAttributes()
+        .putValue(Attributes.Name.MAIN_CLASS.toString(), ForkMain.class.getName());
+    manifest.getMainAttributes().putValue(GATLING_MANIFEST_VALUE, "true");
```
[the remark this answers is not in the document]
Comment (slandelle) [reply src/main/java/io/gatling/plugin/util/Fork.java:240]: indeed, but not in this PR. Will do.

File: src/main/java/io/gatling/plugin/util/Fork.java (hunk 1 above)
Comment (slandelle) [reply src/main/java/io/gatling/plugin/util/Fork.java:248]: I don't think so. It's not a resources, it's a way to move forward in the zip stream.

File: src/main/java/io/gatling/plugin/util/Fork.java (hunk 2 above)
Comment (slandelle) [reply src/main/java/io/gatling/plugin/util/Fork.java:257]: same as above

File: src/main/java/io/gatling/plugin/util/ForkMain.java (hunk 4 above)
Comment (slandelle) [reply src/main/java/io/gatling/plugin/util/ForkMain.java:42]: Yeah, it's not really a file at this point but a ClassLoader resource and an entry in the zip/jar... I'll rename.

File: src/main/java/io/gatling/plugin/util/Fork.java (hunk 3 above)
Comment (slandelle) [reply src/main/java/io/gatling/plugin/util/Fork.java:254]: Anything works, actually. Going with the simple `\n` then.

**Review by Isammoc — APPROVED** [review]

**Review by guilgaly — APPROVED** [review]

**Pull request merged**